File: frontend/src/microservices/order/objects/SupplierReviewDecisionResponse.ts
// 本文件定义 order 模块的 `SupplierReviewDecisionResponse`，作为响应数据并提供 JSON 编解码。

export type SupplierReviewDecisionResponse = {
  decision: string
  reason: string | null
  decidedAt: string
  managerId: string
}
export const supplierReviewDecisionResponseFromJson = (json: string): SupplierReviewDecisionResponse =>
  JSON.parse(json) as SupplierReviewDecisionResponse

export const supplierReviewDecisionResponseToJson = (value: SupplierReviewDecisionResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/operations/overall/objects/SupplierReviewDecisionResponse.scala
// 本文件定义任务列表里使用的供应商审核决策响应对象。
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class SupplierReviewDecisionResponse(decision: String, reason: Option[String], decidedAt: Option[String], managerId: Option[String])
object SupplierReviewDecisionResponse:
  given sourceEncoder: Encoder[SupplierReviewDecisionResponse] = deriveEncoder
  given sourceDecoder: Decoder[SupplierReviewDecisionResponse] = deriveDecoder
